src/sessionManager.ts

```diff
@@ -213,20 +245,18 @@ export default function SessionManager(
         if (!mpInstance._Store.webviewBridgeEnabled) {
             const persistence = mpInstance._Persistence.getPersistence();
 
-            if (!mpInstance._Store.sessionId && persistence) {
-                if (persistence.sid) {
-                    mpInstance._Store.sessionId = persistence.sid;
-                } else {
-                    self.startNewSession();
-                }
+            if (
+                persistence?.gs &&
+                persistence?.gs.sid &&
+                mpInstance._Store.sessionId !== persistence.gs.sid
+            ) {
+                mpInstance._Store.sessionId = persistence.gs.sid;
+            }
+
+            // TODO: Make this a store method
```

Comment:
Can you clarify this TODO?  Are you arguing that `startNewSession` should be on the store and not on the session module?

Reply:
I think the answer the the above is yes.  However I think there's a lot happening in this method that isn't just store related.  Maybe there is a way to keep the session logic in session, and at the end, create a new method about updating the store's session attributes.

Reply:
I probably should have put this comment before the `startNewSession` method. My thought is that we're doing a lot of session related activities within the context of a store. Part of me thinks that we should split out the Session related persistence/store attributes into a seaprate Session object, so that the Session Manager can simply manage Sessions. As we've discussed, the store holds a lot of data about the state of the SDK, but a lot of unrelated modules directly modify the state which makes it hard to predict what the current state of the SDK is, so I thought a quick path to tidying things up would be to create a `startNewSession` method in the store to clean some things up.